snapshots/output/syntax/src/class.ts

```diff
@@ -0,0 +1,49 @@
+  export class Class {
+//             ^^^^^ reference example 1.0.0 src/class.ts/Class#
+    public classProperty: string
+//         ^^^^^^^^^^^^^ definition example 1.0.0 src/class.ts/Class#classProperty.
+    constructor(constructorParam: string) {
+//              ^^^^^^^^^^^^^^^^ definition example 1.0.0 src/class.ts/Class#<constructor>().(constructorParam)
+      this.classProperty = constructorParam
+//         ^^^^^^^^^^^^^ reference example 1.0.0 src/class.ts/Class#classProperty.
+//                         ^^^^^^^^^^^^^^^^ reference example 1.0.0 src/class.ts/Class#<constructor>().(constructorParam)
+    }
+    public method(methodParam: string): string {
+//         ^^^^^^ definition example 1.0.0 src/class.ts/Class#method().
+//                ^^^^^^^^^^^ definition example 1.0.0 src/class.ts/Class#method().(methodParam)
+      return this.privateMethod(methodParam)
+//                ^^^^^^^^^^^^^ reference example 1.0.0 src/class.ts/Class#privateMethod().
+//                              ^^^^^^^^^^^ reference example 1.0.0 src/class.ts/Class#method().(methodParam)
+    }
+    public static staticMethod(methodParam: string): string {
+//                ^^^^^^^^^^^^ definition example 1.0.0 src/class.ts/Class#staticMethod().
```

Comment:
Typescript allows a class to have an instance method with the same signature as a static method; if that happens, we'll run into an ambiguity because the syntax is identical (`Class#method()`), right?